### PR TITLE
adds proj=True argument to plot_raw

### LINKED ISSUE #4190: raw.plot() silently applies projectors that are disabled
**how it works now**

- inactive projectors are automatically applied during `raw.plot()`
- a small "proj" button appears in the bottom right of the plot window
- Clicking the proj button shows a list of projectors, each with a large "X" next to it.  

**how it should work**

- if a raw object has projectors that are ***not activated***, `raw.plot()` should plot the ***untransformed data***
- the big "X" marks should be replaced with check marks, or something else that is less ambiguous (maybe click on the projector names to make them black vs. grey, analogous to bad channels?).  It is quite plausible to interpret "X" as either "selected" (i.e., "active") or as "crossed off" (i.e., "not active").
- at a minimum, the whole phenomenon of the "proj" button should be documented in the `raw.plot()` docstring.
- optionally, `raw.plot()` could get a `proj` argument that lets the projectors get applied (non-interactively) before plotting.  The simplest version might accept only "all or none" booleans; a more complex implementation might accept a list of indices into `raw.projs` indicating which projections to activate.

### PR DESCRIPTION
**Pull request by drammock**
closes #4190.  @Eric89GXL are you reviewing this one or should I ping someone else?

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Changes look reasonable but Travis is not happy:

https://travis-ci.org/mne-tools/mne-python/jobs/223691431

I'll try it tomorrow. @jaeilepp feel free to have a look to, it changes some aesthetics

File: mne/viz/raw.py [hunk 1]
```diff
@@ -178,6 +178,11 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
         then excessive values are not shown, creating gaps in the traces.
     show_first_samp : bool
         If True, show time axis relative to the ``raw.first_samp``.
+    proj : bool
+        Whether to apply projectors prior to plotting. Individual projectors
+        can be enabled/disabled interactively (see Notes). Note also that this
+        argument only affects the plot; use ``raw.apply_proj()`` to modify the
+        data stored in the Raw object.
```
Comment (jaeilepp) [inline mne/viz/raw.py]: Document the default.

File: mne/viz/utils.py [hunk 2]
```diff
@@ -252,7 +252,9 @@ def _toggle_proj(event, params):
             if not b and p['active']:
                 bools[bi] = True
     else:
-        bools = [True] * len(params['projs'])
+        proj = (True if 'apply_proj' not in params.keys() else
+                params['apply_proj'])
```
Comment (jaeilepp) [inline mne/viz/utils.py]: -> params.get('apply_proj', True)

File: mne/viz/utils.py [hunk 3]
```diff
@@ -385,19 +387,24 @@ def _draw_proj_checkbox(event, params, draw_current_state=True):
 
     labels = [p['desc'] for p in projs]
     actives = ([p['active'] for p in projs] if draw_current_state else
-               [True] * len(params['projs']))
+               [params['apply_proj']] * len(projs))
 
     width = max([len(p['desc']) for p in projs]) / 6.0 + 0.5
```
Comment (jaeilepp) [inline mne/viz/utils.py]: While you're at it, please add a minimum width to the figure. If the proj names are short it cuts off the title. I tested and seems like 4 is a good default size so change this line to 
``width = max([max([len(p['desc']) for p in projs]) / 6.0 + 0.5, 4.])``

**Comment by Eric89GXL** [comment]
Works great over here, will merge once the CIs come back happy

**Comment by Eric89GXL** [comment]
Flake:

https://travis-ci.org/mne-tools/mne-python/jobs/224028374#L3162

You can test locally with `make flake` or `make pep`

**Comment by drammock** [comment]
arggh, was editing in a different editor than usual, the ruler was at 80 instead of 79.  Should be fixed now.

**Comment by Eric89GXL** [comment]
Thanks @drammock

**Pull request merged**